### PR TITLE
implemented NOA filings + Colin-only filings

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#4210

*Description of changes:*
This PR contain multiple commits that can be reviewed separately.

1. Pre-implementation Cleanup:
- renamed an enum mixin method
- renamed some interfaces for consistency
- removed unneeded enum declarations
- renamed todoItems and historyItems to distinguish them from filings
- misc cleanup

2a. Implementation:
- added ColinFiling component
- added NoticeOfAlteration component
- added history item subtitle
- added history item action button labels
- added NOA and Colin filing handling
- added types to filings, tasks, history items and task items
- added new entity names and types
- added new filing name and type
- updated enum mixin
- added Alteration interface
- added Filing interface
- added History Item interface
- added Task Item interface
- renamed interfaces consistently
- added/updated unit tests

2b. Improvements + Unit Tests:
- added Business interface
- added Header interface
- added unit tests for NOAs
- added unit test for Colin filings

2c. Text Updates per Janis/Kaine:
- updated NOA subtype labels
- updated NOA component text
- updated NOA title + mixin
- updated unit tests

2d. More Updates per Updated Schema:
- removed obsolete alteration object
- renamed alteration object per schema
- added alteration interfaces
- removed obsolete unit test
- updated unit tests

2e. More Updates and Improvements:
- updated expand button label per Janis
- added props to Header interface
- updated unit tests

2f. Preparation for New Entity Types (Corp Types):
- renamed EntityTypes -> LegalTypes (consistent with BE)
- renamed EntityNames -> LegalNames
- updated/added legal types and names consistent with Colin
- added NameRequestTypes to handle differences between Namex and Lear
- added "filed" label method with conditionals
- misc cleanup

2g. added missing semi-colons

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/interfaces/alteration-interface.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { EntityTypes } from '@/enums'
+
+/** A filing's alteration object from the API. */
+export interface AlterationIF {
+  alterCorpType?: {
+    corpType: EntityTypes;
+  }
+  alterResolutions?: any;
```
Comment (lekshmimallika-aot) [inline src/interfaces/alteration-interface.ts:8]: There is no sub structure like alterResolutions. We have an array of resolution dates within alterShareStructure

File: src/interfaces/alteration-interface.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  alterCorpName?: any;
+  alterNameTranslation?: any;
```
Comment (lekshmimallika-aot) [inline src/interfaces/alteration-interface.ts:10]: alterNameTranslations

File: src/interfaces/alteration-interface.ts (hunk 1 above)
Comment (severinbeauvais) [reply src/interfaces/alteration-interface.ts]: Fixed. Please review latest code.

File: src/interfaces/alteration-interface.ts (hunk 2 above)
Comment (severinbeauvais) [reply src/interfaces/alteration-interface.ts]: Fixed. Please review latest code.

File: src/interfaces/alteration-interface.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { EntityTypes } from '@/enums'
+
+export interface AlterCorpTypeIF {
+  corpType: EntityTypes;
+  [propName: string]: any; // excess properties
```
Comment (cameron-freshworks) [inline src/interfaces/alteration-interface.ts:5]: This is pretty interesting!

File: src/interfaces/business-interface.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { FilingTypes } from '@/enums'
+
+/** A filing's business object from the API. */
+export interface BusinessIF {
+  identifier: string,
+  legalName: string,
+  legalType?: FilingTypes,
```
Comment (cameron-freshworks) [inline src/interfaces/business-interface.ts:7]: Maybe missed some semicolons here.

File: src/interfaces/alteration-interface.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+/** A filing's alteration object from the API. */
+export interface AlterationIF {
+  provisionsRemoved?: boolean;
+  alterCorpType?: AlterCorpTypeIF
+  alterCorpName?: AlterCorpNameIF
+  alterNameTranslations?: AlterNameTranslationsIF
+  alterShareStructure?: AlterShareStructureIF
```
Comment (cameron-freshworks) [inline src/interfaces/alteration-interface.ts:32]: Were the semicolons intentionally left out here?

Comment (severinbeauvais) [reply src/interfaces/alteration-interface.ts:32]: Fixed in next commit.

File: src/interfaces/business-interface.ts (hunk 4 above)
Comment (severinbeauvais) [reply src/interfaces/business-interface.ts:7]: Fixed in next commit.

**Comment by cameron-freshworks** [comment]
Not sure i understand the SonarCloud code warning.

**Comment by severinbeauvais** [comment]
> Not sure i understand the SonarCloud code warning.

I looked into it. Apparently I caused 74 duplicated lines and 2 duplicate blocks -- both in the updated LegalNames / LegalTypes enums. Sure enough, these enums have the same set of constants, but different values assigned in each file. I wouldn't want to write this code any differently so I'm inclined to ignore this "failure" :)

**Comment by cameron-freshworks** [comment]
> > Not sure i understand the SonarCloud code warning.
> 
> I looked into it. Apparently I caused 74 duplicated lines and 2 duplicate blocks -- both in the updated LegalNames / LegalTypes enums. Sure enough, these enums have the same set of constants, but different values assigned in each file. I wouldn't want to write this code any differently so I'm inclined to ignore this "failure" :)

Absolutely, no problems there!
The one in question i was referring too was suggesting renaming the 'NamexRequestMixin'.. but i didn't quite get why.

**Comment by severinbeauvais** [comment]
> The one in question i was referring too was suggesting renaming the 'NamexRequestMixin'.. but i didn't quite get why.

Because the subject mixin has nothing to do with Namex. We get the NR data from legal-api, and we don't care what it connects to for the data (namex or nro or whatever). This mixin is purely working with a NR.

**Comment by cameron-freshworks** [comment]
> > The one in question i was referring too was suggesting renaming the 'NamexRequestMixin'.. but i didn't quite get why.
> 
> Because the subject mixin has nothing to do with Namex. We get the NR data from legal-api, and we don't care what it connects to for the data (namex or nro or whatever). This mixin is purely for working with a NR.

I see, so maybe it is a valid warning!

**Comment by severinbeauvais** [comment]
> > > The one in question i was referring too was suggesting renaming the 'NamexRequestMixin'.. but i didn't quite get why.
> > 
> > 
> > Because the subject mixin has nothing to do with Namex. We get the NR data from legal-api, and we don't care what it connects to for the data (namex or nro or whatever). This mixin is purely for working with a NR.
> 
> I see, so maybe it is a valid warning!

For other reviewers... I got this all wrong. Cam was asking about the code smell that SonarCloud reported. It was because I renamed the subject mixin file but I forgot to rename the exported class! (SC just recommended that filename and export should match.) So I've now fixed this.

**Pull request merged**